In [13]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [14]:
import os

project_path = '/content/drive/MyDrive/Dist-Abl-PRL-All-Exs-ETTH1'
os.chdir(project_path)

print("Current Directory:", os.getcwd())

Current Directory: /content/drive/MyDrive/Dist-Abl-PRL-All-Exs-ETTH1


In [16]:
!ls -la experiments/exp4b_order_input_mentor_version/e4b_order_ip_mentor_vr_ph1.sh

-rw------- 1 root root 10956 Apr  5 18:58 experiments/exp4b_order_input_mentor_version/e4b_order_ip_mentor_vr_ph1.sh


In [17]:
import os

# Set your Informer data directory
informer_path = f'{project_path}/Informer2020-original'
data_dir = f'{informer_path}/data/ETT'
os.makedirs(data_dir, exist_ok=True)

# File path
data_file = f'{data_dir}/ETTh1.csv'

# Download ETTh1 dataset
if not os.path.exists(data_file):
    print("Downloading ETTh1.csv...")
    os.system(
        f'wget -q "https://raw.githubusercontent.com/zhouhaoyi/ETDataset/main/ETT-small/ETTh1.csv" '
        f'-O "{data_file}"'
    )

# Verify
if os.path.exists(data_file):
    print(f"✅ Download successful: {data_file}")
else:
    print("❌ Download failed.")

✅ Download successful: /content/drive/MyDrive/Dist-Abl-PRL-All-Exs-ETTH1/Informer2020-original/data/ETT/ETTh1.csv


In [18]:
# Fix np.Inf -> np.inf in tools.py (removed in NumPy 2.0)
tools_path = f'{project_path}/Informer2020-original/utils/tools.py'

with open(tools_path, 'r') as f:
    content = f.read()

content_fixed = content.replace('np.Inf', 'np.inf')

with open(tools_path, 'w') as f:
    f.write(content_fixed)

print("✅ Fixed np.Inf -> np.inf in utils/tools.py")

✅ Fixed np.Inf -> np.inf in utils/tools.py


In [20]:
!bash experiments/exp4b_order_input_mentor_version/e4b_order_ip_mentor_vr_ph1.sh

Experiment 4b — Phase 1: Legendre Delta in Value Matrix
Start: Sun Apr  5 07:00:19 PM UTC 2026

Config (fixed — no alpha sweep):
  Modification  : Δp_i = P_i - P_{i-1} in attention value
  Input embed   : value + temporal only (no ordering added)
  Extra file    : ordering_operator_positional.py (required)
  Seed          : 2021 (single)
  Pred lengths  : 96, 192
  Total runs    : 2

Reference — Best previous (Exp1-Pre, alpha=1.0):
  pred_len=96  : MSE=?, MAE=?  (fill from your logs)
  pred_len=192 : MSE=?, MAE=?  (fill from your logs)

Copying experiment files...
Files copied:
  ✓ __init__.py
  ✓ attn.py
  ✓ embed.py
  ✓ encoder.py
  ✓ decoder.py
  ✓ model.py
  ✓ ordering_operator_positional.py  (extra — required)

------------------------------------------------------------
RUN 1/2: exp4b_ph1_ETTh1_legdelta_pred96_seed2021
Start: Sun Apr  5 07:00:24 PM UTC 2026
Args in experiment:
Namespace(model='informer', data='ETTh1', root_path='./data/ETT/', data_path='ETTh1.csv', features='M', 

## What to note from Phase 1:

- Both runs completed cleanly, early stopping at epoch 4 both times — validation loss was rising from epoch 1, which means the model is not learning well with this modification  

- The Legendre delta in the value matrix appears to be hurting rather than helping — the attention is aggregating positional deltas instead of semantic content, which likely destroys the signal  

- Without your Exp1-Pre reference numbers I cannot make the official go/no-go call, but the pattern looks like a negative result  

- You must check: What were your Exp1-Pre alpha=1.0 MSE numbers at pred_len=96 and 192? Share them and I'll give you the definitive decision before you waste compute on Phase 2.

In [21]:
!bash experiments/exp4b_order_input_mentor_version/e4b_order_ip_mentor_vr_ph2.sh

Experiment 4b — Phase 2: Legendre Delta in Value Matrix
Start: Sun Apr  5 07:20:24 PM UTC 2026

Phase 1 recap (seed=2021):
  pred_len=96  : MSE=0.7447, MAE=0.6672
  pred_len=192 : MSE=0.8538, MAE=0.7253
  Early stopping at epoch 4 in both runs

Phase 2 config:
  Modification  : Δp_i = P_i - P_{i-1} in attention value
  Input embed   : value + temporal only (no ordering added)
  Extra file    : ordering_operator_positional.py (required)
  Seeds         : 2021, 2022, 2023
  Pred lengths  : 48, 96, 192
  Total runs    : 9 (2 already done in Phase 1 — skipped)

Reference — Best previous (Exp1-Pre, alpha=1.0):
  pred_len=48  : MSE=?  MAE=?  (fill from your exp1 logs)
  pred_len=96  : MSE=?  MAE=?  (fill from your exp1 logs)
  pred_len=192 : MSE=?  MAE=?  (fill from your exp1 logs)

Copying experiment files...
Files copied:
  ✓ __init__.py
  ✓ attn.py
  ✓ embed.py
  ✓ encoder.py
  ✓ decoder.py
  ✓ model.py
  ✓ ordering_operator_positional.py  (extra — required)

-----------------------------